src/runtime/server/server.ts

```diff
@@ -259,28 +180,34 @@ interface ServerFactory {
     schema: GraphQL.GraphQLSchema
     plugins: Plugin.RuntimeContributions[]
     contextContributors: ContextContributor<any>[]
-    settings: App.Settings
   }): Promise<void>
   stop(): Promise<void>
   express: ExpressInstance
+  settings: {
+    change(settings: ServerSettings.SettingsInput): void
+    data: ServerSettings.SettingsData
+  }
 }
 
 export function create(): ServerFactory {
   const express = createExpress()
   let server: BaseServer | null = null
+  const state = {
+    settings: ServerSettings.defaultSettings,
```

Comment:
Not sure this should be so eager. For example/for one thing, tests where we want to manipulate the env.

Generally I'd just be very careful about anything that is stateful to the module, avoid where possible. Seems like we can here.

```suggestion
    settings: ServerSettings.defaultSettings(),
```